lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -1873,89 +1871,186 @@ void TFDeabstraction::checkAndCanonicalizeAttributes() {
       if (!TFStrictDeabstraction)
         continue;
 
-      // Use the constants we just computed to substitute into parameter values
-      // if we don't already have them.
-      // TODO: this should eventually create a new SILInstruction to represent
-      // the graph operation instead of using SIL instructions to represent the
-      // constants.
-      bool isError = false;
-      SILBuilder B(opInfo->inst);
-      for (unsigned i = 0, e = opInfo->operandClasses.size(); i != e; ++i) {
-        // Ignore input operands.
-        if (opInfo->isInput(i))
-          continue;
+      formGraphOp(*opInfo, constants);
```

Comment:
how do we plan to use/consume `constants` here, or will there be other call-sites that will consume this output param?

Reply:
constants contains all the attribute values, the next patch turns them into arguments to graph_op.